src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -0,0 +1,169 @@
+        /// </summary>
+        public bool RunPreviewIsChecked
+        {
+            get
+            {
+                return _runPreviewIsChecked;
+            }
+            set
+            {
+                _runPreviewIsChecked = value;
+                RaisePropertyChanged("RunPreviewIsChecked");
+            }
+        }
+
+        /// <summary>
+        /// LanguagesList property containt the list of all the languages listed in: https://wiki.autodesk.com/display/LOCGD/Dynamo+Languages
+        /// </summary>
+        public ObservableCollection<string> LanguagesList
+        {
+            get
+            {
+                return _languagesList;
+            }
+            set
+            {
+                _languagesList = value;
+                RaisePropertyChanged("LanguagesList");
+            }
+        }
+
+        /// <summary>
+        /// FontSizesList contains the list of sizes for fonts defined (the ones defined are Small, Medium, Large, Extra Large)
+        /// </summary>
+        public ObservableCollection<string> FontSizeList
+        {
+            get
+            {
+                return _fontSizeList;
+            }
+            set
+            {
+                _fontSizeList = value;
+                RaisePropertyChanged("FontSizeList");
+            }
+        }
+
+        /// <summary>
+        /// NumberFormatList contains the list of the format for numbers, right now in Dynamo has the next formats: 0, 0.0, 0.00, 0.000, 0.0000
+        /// </summary>
+        public ObservableCollection<string> NumberFormatList
+        {
+            get
+            {
+                return _numberFormatList;
+            }
+            set
+            {
+                _numberFormatList = value;
+                RaisePropertyChanged("NumberFormatList");
+            }
+        }
+
+        /// <summary>
+        /// The PreferencesViewModel constructor basically initialize all the ItemsSource for the corresponding ComboBox in the View (PreferencesView.xaml)
+        /// </summary>
+        public PreferencesViewModel()
+        {
+
+            string languages = Wpf.Properties.Resources.PreferencesWindowLanguages;
+            LanguagesList = new ObservableCollection<string>(languages.Split(','));
+            SelectedLanguage = languages.Split(',').First();
+
+            string fontSizes = Wpf.Properties.Resources.PreferencesWindowFontSizes;
+            FontSizeList = new ObservableCollection<string>(fontSizes.Split(','));
+            SelectedFontSize = fontSizes.Split(',')[1];
+
+            string numberFormats = Wpf.Properties.Resources.PreferencesWindowNumberFormats;
+            NumberFormatList = new ObservableCollection<string>(numberFormats.Split(','));
+            SelectedNumberFormat = numberFormats.Split(',').First();
+
+            //By Default the Default Run Settings radio button will be in Manual
+            RunSettingsIsChecked = true;
```

Comment:
These are fine for now, we may need to consider ahead what needs to happen when user click `Save`, by looking at the design, all the changes take effect after clicking on `Save` which lead to some overhead..

Reply:
@QilongTang 
I can see that all the setting are saved in a file named DynamoSettings.xml, located in the AppData folder, then I think that when we implement the functionality for the "Save Changes" button, we will need to set all the default options first and then load the options located in the xml file.

Reply:
Thanks, the preference setting should already be deserialized when launching Dynamo at this point. We will need to keep it in sync with the setting here when we implement the actual functionalities. Thanks for the brain storming.